### PR TITLE
Flyway Lobby DB Migrations

### PR DESCRIPTION
**Pull request by DanVanAtta**
- Move migration files to a 'lobby-db' subproject
- Rename migration files to fit flyway default naming
- Remove transaction block in migration files, flyway
  opens a DB transaction per migration file automatically
- Add build file to 'lobby-db' with flyway plugin to be able to run flyway
- Create convenience wrapper scripts for dev to recreate DB and run flyway
- Create scripts for running on prod, should be self documenting when run
- Add a sub-project README with documentation

Liquibase and DB migrations were brought up and discussed in: #2754, #2356

### CODE REVIEW

File: lobby-db/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+### Prod Deployments
+- Clone the project to lobby server
+- cd to the `lobby-db` project folder
+
+- run: `./prod_check_flyway.sh`
+  - Follow prompts and add user+password to creds file if not already done.
+
+First time you should see output like this:
+```
+dan@dan-desk:~/work/triplea/lobby-db$ ./prod_run_flyway.sh
```
Comment (ron-murhammer) [inline lobby-db/README.md]: Probably should remove references to your login.

**Comment by ron-murhammer** [comment]
Other than removing references to your login, it looks alright to me.

File: lobby-db/README.md (hunk 1 above)
Comment (DanVanAtta) [reply lobby-db/README.md]: Point taken, this is the example for prod deployment, not a developer deploy. I pushed 441ca4c which would update the example to more 'prod-like.'

File: lobby-db/.include/prod_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+     ;;
+  flywayInfo)
+     ;;
+   *)
+     echo "Param must be a one of 'flywayInfo' or 'flywayMigrate', it was: '$TASK'"
+     exit 1
+esac
+
+
+CREDS_FILE=~/lobby.db.creds
```
Comment (ssoloff) [inline lobby-db/.include/prod_deploy.sh]: Probably want to create this file with a leading period so it doesn't pollute a user's home directory listing.  Also, you may want to consider adding a _triplea-_ prefix (or placing it in the _~/triplea_ folder) so it's obvious it is related to TripleA in a sea of otherwise unrelated files.

File: lobby-db/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
+## lobby-db
+
+A project that contains the migration scripts for the lobby postgres database.
+
+### Prod Deployments
```
Comment (ssoloff) [inline lobby-db/README.md]: Did you happen to try this on a copy of the production DB?  I'm curious how Flyway handles the fact that migration scripts 1.00-1.02 have already been applied, but there is currently no Flyway history table that records that information.  Will Flyway attempt to re-apply those scripts and just error out?

File: lobby-db/.include/prod_deploy.sh (hunk 2 above)
Comment (DanVanAtta) [reply lobby-db/.include/prod_deploy.sh]: Shared account of TripleA user should have almost no files, pollution should not be a problem. It's a good consideration, for a developer rig I'd agree, for prod server I think the explicit visibility is also a benefit.

The folder namespacing is not a bad suggestion, but it either requires us to create the folder or to install lobby first. For now I'd like to start simple. After first prod deployment I expect us to update some of these scripts to match what really works out in the end.

File: lobby-db/README.md (hunk 3 above)
Comment (DanVanAtta) [reply lobby-db/README.md]: Out of scope for this documentation, but to satisfy curiosity it will error out:
```
> Error occurred while executing flywayMigrate
  Found non-empty schema(s) "public" without schema history table! Use baseline() or set baselineOnMigrate to true to initialize the schema history table.
```

**Comment by DanVanAtta** [comment]
Unassigning @RoiEXLab we do not need a triple review for this; 
@ssoloff did you want more time to spend on review?

**Comment by DanVanAtta** [comment]
Of note, I pushed some fixes for the docker configuration and some branching logic to have the lobby-db scripts work on mac. It may be worthwhile having a peek at those @ssoloff, @ron-murhammer

File: lobby-db/README.md (hunk 3 above)
Comment (ssoloff) [reply lobby-db/README.md]: Good to know.  So it would seem that the `flywayBaseline` task is going to need to be run against the production and staging databases before the `flywayMigrate` task.

**Comment by DanVanAtta** [comment]
I'll merge this later today if no further feedback.

File: lobby-db/README.md (hunk 3 above)
Comment (DanVanAtta) [reply lobby-db/README.md]: Exact commands TBD, but otherwise yes  : )
